### PR TITLE
Use const CatalogDatabase reference in QueryContext.

### PR DESCRIPTION
**Pull request by zuyu**
This small PR refactored to use the const reference of `CatalogDatabaseLite` in `QueryContext`, instead of a mutable pointer, as a reconstructed `InsertDestination` in `QueryContext` does not directly update the catalog any more (#98).

### CODE REVIEW

File: query_execution/Foreman.hpp [hunk 1]
```diff
@@ -122,7 +122,7 @@ class Foreman final : public ForemanLite {
    **/
   inline void reconstructQueryContextFromProto(const serialization::QueryContext &proto) {
     query_context_.reset(
-        new QueryContext(proto, catalog_database_, storage_manager_, foreman_client_id_, bus_));
+        new QueryContext(proto, *catalog_database_, storage_manager_, foreman_client_id_, bus_));
```
Comment (pateljm) [inline query_execution/Foreman.hpp]: Why not `&catalog_database_`?

File: relational_operators/tests/AggregationOperator_unittest.cpp [hunk 2]
```diff
@@ -260,7 +260,7 @@ class AggregationOperatorTest : public ::testing::Test {
 
     // Set up the QueryContext.
     query_context_.reset(new QueryContext(query_context_proto,
-                                          db_.get(),
+                                          *db_,
```
Comment (pateljm) [inline relational_operators/tests/AggregationOperator_unittest.cpp]: Why not `&db_`? Same comment applies below.

Comment (zuyu) [inline relational_operators/tests/AggregationOperator_unittest.cpp]: `db_` is a `unique_ptr`. So `*db_` returns the reference.

File: query_execution/Foreman.hpp (hunk 1 above)
Comment (zuyu) [inline query_execution/Foreman.hpp]: `catalog_database_` is a pointer, so `* catalog_database_` is a reference.

**Comment by pateljm** [comment]
LTGM. Merging.

**Pull request merged**